os/kernel/debug/mem_leak_checker.c

```diff
@@ -168,6 +170,7 @@ static void fill_hash_table(int *leak_cnt, int *broken_cnt)
 	{
 		node_size = SIZEOF_MM_ALLOCNODE;
 		for (node = leak_checker.heap_start[region]; node < leak_checker.heap_end[region]; node = (struct mm_allocnode_s *)((char *)node + node->size)) {
+			DEBUGASSERT(node->size);
```

Comment:
Is there any reason why you use `DEBUGASSERT` instead of `ASSERT`?

Reply:
No, ASSERT() is correct because these ASSERT checks cause hang. I modified it.